src/app/utils/question.ts

```diff
@@ -0,0 +1,63 @@
+import { db } from "../../../firebase";
+import { Question, QuestionState } from "../types/db";
+import { questionConverter } from "../services/firestore";
+import {
+  doc,
+  DocumentReference,
+  addDoc,
+  collection,
+  CollectionReference,
+  PartialWithFieldValue,
+  updateDoc,
+} from "firebase/firestore";
+
+export const addQuestion = async (
+  question: PartialWithFieldValue<Question>,
+  courseId: string
+) => {
+  try {
+    const questionsColection: CollectionReference = collection(
+      db,
+      `courses/${courseId}/questions`
+    ).withConverter(questionConverter);
+
+    const newQuestion = {
+      title: question.title,
+      id: "",
+      description: question.description,
+      public: question.public,
+      state: QuestionState.Pending,
+      timestamp: question.timestamp,
+      group: question.group,
+      tags: question.tags,
+    };
+
+    const questionDoc = await addDoc(questionsColection, newQuestion);
+
+    const questionId = questionDoc.id;
```

Comment:
Do we need a separate field to keep track of the ID when a doc has an ID?

Reply:
The ID is stored there so we know which question the user is interacting with in the UI